test/end2end/app_test.dart

```diff
@@ -223,7 +225,47 @@ Future<void> createPost(WidgetTester tester) async {
   await tester.drag(find.byType(PostFeed), const Offset(0, 500));
   await tester.pumpAndSettle();
 
-  // Check that the post is displayed
+  // Check that the post is displayed in feed
   expect(find.text(postTitle), findsOneWidget);
   expect(find.text(postDescription), findsOneWidget);
+
+  // Check that the post is displayed in profile page
+  final profilePicture = find.byKey(AppTopBar.profilePictureKey);
+  await tester.tap(profilePicture);
+  await tester.pumpAndSettle();
+  expect(find.text(postTitle), findsOneWidget);
+  expect(find.text(postDescription), findsOneWidget);
+}
+
+/// Delete a post
+Future<void> deletePost(WidgetTester tester) async {
+  expect(find.byType(ProfilePage), findsOneWidget);
+
+  // Check that the post card is displayed
+  final postCard = find.byKey(ProfileInfoCard.infoCardKey);
+  expect(postCard, findsOneWidget);
+
+  // Find the delete button on card
+  final deleteButton = find.byKey(ProfileInfoCard.deleteButtonCardKey);
+  expect(deleteButton, findsOneWidget);
+
+  await tester.tap(deleteButton);
+  await tester.pumpAndSettle(delayNeededForAsyncFunctionExecution);
+
+  // Can't find post anymore
+  const postTitle = "I like turtles";
+  const postDescription = "Look at them go!";
+  expect(find.text(postTitle), findsNothing);
+  expect(find.text(postDescription), findsNothing);
+
```

Comment:
I think that we could also check that the post card is not there anymore in case of another action happening (like a renaming action instead of a deletion) 😄 .

```suggestion
  // Can't find post anymore
  const postTitle = "I like turtles";
  const postDescription = "Look at them go!";
  expect(find.text(postTitle), findsNothing);
  expect(find.text(postDescription), findsNothing);

  // Check that the post card is not displayed anymore
  expect(postCard, findsNothing);

```

Reply:
Good idea! Committed your suggestion in dd6a29abdcccd23fd33d6605454a9209101d8639, thanks :)